src/NLog/Targets/TargetWithContext.cs

```diff
@@ -276,6 +277,104 @@ protected bool ShouldIncludeProperties(LogEventInfo logEvent)
             return combinedProperties ?? new Dictionary<string, object>(StringComparer.Ordinal);
         }
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="logEvent"></param>
+        /// <returns></returns>
+        protected IEnumerable<KeyValuePair<string, object>> GetAllPropertiesList(LogEventInfo logEvent)
+        {
+            if (EstimateAllPropertiesCount(logEvent, out var logEventProperties, out var scopeContextProperties, out var gdcNames, out int propertiesCount) > 0)
+            {
+                List<KeyValuePair<string, object>> combinedProperties = new List<KeyValuePair<string, object>>(propertiesCount);
```

Comment:
@snakefoot 
1) _Any reason for allocating List, instead of using yield-engine ?_
   I've rewritten method to use yield return

2) _Seems we are missing the ContextProperties ?_
   Yep... now fixed

3) _Believe you are missing logic from CombineProperties..._
   I will look at this tomorrow

Reply:
Notice I will be on vacation next week, so there will be less feedback.